packages/btcindexer/src/btcindexer.ts

```diff
@@ -30,12 +30,35 @@ interface BlockRecord {
 	block_height: number;
 }
 
-interface Storage {
+export interface Storage {
 	d1: D1Database; // SQL DB
 	blocksDB: KVNamespace;
 	nbtcTxDB: KVNamespace;
 }
 
+export type NbtcTxStatus = "confirming" | "finalized" | "minted" | "failed" | "reorg";
+
+export interface TransactionStatusResponse {
+	btc_tx_id: string;
+	status: NbtcTxStatus;
+	block_height: number | null;
+	confirmations: number;
+	sui_recipient: string;
+	amount_sats: number;
+}
+
+interface NbtcTxD1Row {
+	tx_id: string;
+	block_hash: string;
+	block_height: number | null;
+	vout: number;
+	sui_recipient: string;
+	amount_sats: number;
+	status: string;
```

Comment:
this should be enum

Reply:
we have the `NbtcTxStatus` type that is typesafe